CIScripts/Tests/SNATTest.ps1

```diff
@@ -1,3 +1,350 @@
+    $SNATDiskDir = "C:\snat-vm-image"
+    $SNATDiskPath = $SNATDiskDir + "\snat-vm-image.vhdx"
+    $SNATVMDir = "C:\snat-vm"
+
+    # Some random GUID for vrouter_hyperv.py
+    $NameLen = 14
+    $SNAT_GUID = "d8cf77cb-b9b1-4d7d-ad3c-ef54f417cb5f"
+    $SNATVMName = "contrail-wingw-$SNAT_GUID"
+    $SNATLeftName = "int-$SNAT_GUID".Substring(0, $NameLen)
+    $SNATRightName = "gw-$SNAT_GUID".Substring(0, $NameLen)
+    $SNATVethName = "veth-$SNAT_GUID".Substring(0, $NameLen)
+
+    New-MgmtSwitch -Session $Session -MgmtSwitchName $SNATMgmtSwitchName
+
+    $SNATVeth = New-RoutingInterface -Session $Session `
+        -SwitchName $TestConfiguration.VMSwitchName `
+        -Name $SNATVethName `
+        -IPAddress $SNATConfiguration.VethIP
+
+    # TODO: Remove `ForwardingMAC` when Agent is functional
+    New-SNATVM -Session $Session `
+        -VmDirectory $SNATVMDir -DiskPath $SNATDiskPath `
+        -MgmtSwitchName $SNATMgmtSwitchName `
+        -VRouterSwitchName $TestConfiguration.VMSwitchName `
+        -RightGW $SNATConfiguration.GatewayIP `
+        -LeftGW $SNATConfiguration.ContainerGatewayIP `
+        -ForwardingMAC $SNATVeth.MacAddress `
+        -GUID $SNAT_GUID
+
+    Write-Host "Extracting adapters data..."
+    $PhysicalAdapter = Get-RemoteNetAdapterInformation -Session $Session -AdapterName $TestConfiguration.AdapterName
+    $HNSAdapter = Get-RemoteNetAdapterInformation -Session $Session -AdapterName HNSTransparent
+    $SNATLeft = Get-RemoteVMNetAdapterInformation -Session $Session -VMName $SNATVMName -AdapterName $SNATLeftName
+    $SNATRight = Get-RemoteVMNetAdapterInformation -Session $Session -VMName $SNATVMName -AdapterName $SNATRightName
+    Write-Host "Extracting adapters data... DONE"
+
+    $SNATLeftVif = 102
+    $SNATLeftNh = 102
+    $SNATRightVif = 103
+    $SNATRightNh = 103
+
+    Write-Host "Setting up routing rules..."
+    $Res = Invoke-Command -Session $Session -ScriptBlock {
+        route add $Using:SNATConfiguration.GatewayIP mask 255.255.255.255 $Using:SNATConfiguration.VethIP "if" $Using:SNATVeth.IfIndex | Out-Null
+        $LASTEXITCODE
+    }
+    if ($Res -ne 0) {
+        throw "Setting up routing rules... FAILED"
+    }
+    Write-Host "Setting up routing rules... DONE"
+
+    Write-Host "Setting up ARP rule for GW-veth forwarding..."
+    $Res = Invoke-Command -Session $Session -ScriptBlock {
+        netsh interface ipv4 add neighbor $Using:SNATVeth.IfIndex $Using:SNATConfiguration.GatewayIP $Using:SnatRight.MacAddressWindows | Out-Null
+        $LASTEXITCODE
+    }
+    if ($Res -ne 0) {
+        throw "Setting up ARP rule for GW-veth forwarding... FAILED"
+    }
+    Write-Host "Setting up ARP rule for GW-veth forwarding... DONE"
+
+    Write-Host "Enable forwarding on the HNSTransparent adapter..."
+    $Res = Invoke-Command -Session $Session -ScriptBlock {
+        netsh interface ipv4 set interface $Using:HNSAdapter.IfIndex forwarding="enabled" | Out-Null
```

Comment:
Hmm, in the future we will need to rethink if we should remove the responsibility if enabling the forwarding extension from Docker Driver. Because right now, Driver initializes the dataplane vswitch, but Agent should presumably init the management vswitch, which is kinda weird. We will need to discuss it probably.